docs/training-run/index.txt

```diff
@@ -4,106 +4,255 @@
+
+Because project directories can include large artifacts that should not be packaged as part of the
+model definition, including data sets or compiled binaries, users can specify a ``.detignore`` file
+at the top level, which lists the file paths to be omitted from the model definition. The
+``.detignore`` file uses the same syntax as `.gitignore <https://git-scm.com/docs/gitignore>`__.
+Byte-compiled Python files, including ``.pyc`` files and ``__pycache__`` directories, are always
+ignored.
 
-***********
- Get Ready
-***********
+********************
+ Pre-training Setup
+********************
 
-The trials are created to train the model. The :ref:`searcher <hyperparameter-tuning>` described by
-the experiment configuration defines a set of hyperparameter configurations, each of which
-corresponds to one trial.
+Trials are created to train the model. The :ref:`Hyperparameter Tuning <hyperparameter-tuning>`
+searcher specified in the experiment configuration file defines a set of hyperparameter
+configurations. Each set corresponds to a single trial.
 
-Once the context and configuration for an experiment have reached the master, the experiment waits
-for the scheduler to assign slots to it. If there are no enough idle slots, the master will scale up
-the resource pools automatically.
+After the context and experiment configuration reach the master, the experiment waits for the
+scheduler to assign slots. If there are insufficient idle slots available, the master automatically
+scales up the resource pools.
 
-When a trial is ready to run, the master communicates with the appropriate agent (or agents, in the
-case of :ref:`distributed training <multi-gpu-training>`), which creates containers with the
-configured environment and the submitted training code. We supply a set of default container images
-that are appropriate for many deep learning tasks, but users can also supply a :ref:`custom image
-<custom-docker-images>` if desired. If the specified container images are not existent locally, the
-trial container will fetch the images from the registry.
+When a trial is ready to run, the master communicates with the agent, or :ref:`distributed training
+<multi-gpu-training>` agents, which create(s) containers that have the configured environment and
+training code. A set of default container images applicable to many deep learning tasks is provided,
+but you can also specify a :ref:`custom image <custom-docker-images>`. If the specified container
+images do not exist locally, the trial container fetches the images from the registry.
 
-After starting the containers, each trial runs the ``startup-hook.sh`` that exists in the context
+After starting the containers, each trial runs the ``startup-hook.sh`` script in the context
 directory.
 
-However, this whole process might take very long before each trial starts training.
+The pre-training activity can take a long time before each trial begins training.
```

Comment:
To make sure we don't look too bad, we should point out that in the ideal case this process only takes a couple seconds

Reply:
Done